contracts/tgrade-valset/src/contract.rs

```diff
@@ -532,6 +538,19 @@ fn end_block(deps: DepsMut, env: Env) -> Result<Response, ContractError> {
     // calculate and store new validator set
     let (validators, auto_unjail) = calculate_validators(deps.as_ref(), &env)?;
 
+    // update operators list with info about whether or not they're active validators
+    let ops: Vec<_> = operators()
+        .keys(deps.storage, None, None, Order::Ascending)
+        .collect::<StdResult<_>>()?;
+    for op in ops {
+        let active = validators.iter().any(|val| val.operator == op);
+        operators().update::<_, StdError>(deps.storage, &op, |op| {
```

Comment:
You need to use the diff in the active set to do this. We do this somewhere internally inside `calculate_validators` and update the weight of each operator.

At that point we can efficiently update the active flag (set it when None -> Some(_), unset it when Some(_) -> None, do nothing for all other cases